bindings/go/oci/resolver/url/resolver.go

```diff
@@ -83,6 +86,17 @@ func (resolver *CachingResolver) Ping(ctx context.Context) error {
 		r.Client = resolver.baseClient
 	}
 	if err := r.Ping(ctx); err != nil {
+		errResp := &errcode.ErrorResponse{}
+		if ok := errors.As(err, &errResp); ok {
+			switch errResp.StatusCode {
+			case http.StatusForbidden, http.StatusUnauthorized:
+				// According to the distribution spec the registry might opt for credentials
```

Comment:
We might reference the distribution spec because we discussed this for so long now :D

Reply:
Sure 👍